### DISCUSSION TITLE
How to flashing my custom firmware file to my crazyflie?

### DISCUSSION THREAD

**Discussion by an949na** [bitcraze/discussions#1686] [Q&A]
Hello, bitcraze community guys! I made my custom firmware for my crazyflie 2.1 drone. As a result, I got my own firmware file,(file as .bin) but at cfclient homepage, the firmware files are .zip compressed flie and it contains not only bin file but also elf and lighthouse.bin etc. What can I do more about make firmware zip file? Help please!!

**Comment by ToveRumar** [comment]
Hi @an949na !
Fun that you are creating your own custom FW!

The easiest and most straight forward way to flash it is using our cfloader tool, located in the https://github.com/bitcraze/crazyflie-clients-python repo.

Instructions on how to do this can be found https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/functional-areas/cfloader/. 
But in short you need to run;

`python -m cfloader flash cf2.bin stm32-fw -w radio://0/10/2M/E7E7E7E7E7`

And switch out the binary path and radio address to your corresponding location/address. 

I hope this answered your question!

**Reply by an949na** [reply]
Thank you for your response. I did well by using cfloader flash bash command. And it works well! By the way, I got one remaining question about bitcraze original firmware file. What does Lighthous.bin and works for? I thought containing Lighthouse.bin will be necessary to work drone with lighthouse system. Can you explain for me? thank you!

**Reply by ToveRumar** [reply]
Great!

The lighthouse.bin and aideck_esp.bin are for flashing decks that need firmware. The zip file is a collection of latest releases of all these firmwares so that you will get the latest and greatest on all parts of the system when updating.

I hope this answered your question!
